### PR TITLE
feat: add storybook & designed component

### PR DESCRIPTION
**Pull request by Aylie-Chou**
- add component
  - button
    - pill button
  - text
    - headline
    - paragraph
  - icon
  - divider

- notice
  - icons are stored in gcs, may occur same-origin problem

- how to run storybook on local
  - `yarn storybook`

### CODE REVIEW

File: packages/react-components/src/button/components/pillButton.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  )
+}
+PillButton.propTypes = {
+  text: PropTypes.string,
+  size: PropTypes.oneOf(['S', 'L']),
+  theme: PropTypes.oneOf(['transparent', 'normal', 'photography', 'index']),
+  type: PropTypes.oneOf(['filled', 'outline']),
+  disabled: PropTypes.bool,
+}
+
```
[the remark this answers is not in the document]
Comment (Aylie-Chou) [reply packages/react-components/src/button/components/pillButton.js:84]: default props are defined in L48 - L52.
Would you consider define them with `PillButton.defaultProps` better ?

[the remark this answers is not in the document]
Comment (taylrj) [reply packages/react-components/src/button/components/pillButton.js:84]: It seems defaultProps on function components will eventually get deprecated [[ref](https://github.com/reactjs/rfcs/pull/107)]. 

Using object default values would be better, I'll resolve related comments.

File: packages/react-components/src/icon/topic.stories.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { Topic } from './index'
+
+export default {
+  title: 'Icon/Topic',
+  component: Topic,
+  argTypes: {
+    releaseBranch: {
+      defaultValue: 'master',
+      options: ['master', 'staging', 'preview', 'release'],
```
Comment (taylrj) [inline packages/react-components/src/icon/topic.stories.js:10]: `import releaseBranchConsts from '@twreporter/core/lib/constants/release-branch'` and use the constants ?

File: .circleci/config.yml [hunk 3]
```diff
@@ -184,6 +202,9 @@ workflows:
   version_and_publish:
     jobs:
       - test
+      - storybook:
+          requires:
+            - test
```
Comment (taylrj) [inline .circleci/config.yml:207]: Should we run Chromatic only on `master` branch ?

File: packages/react-components/package.json [hunk 4]
```diff
@@ -9,7 +9,10 @@
     "clean": "make clean",
     "build": "make build",
     "dev": "make dev",
-    "prepublishOnly": "make build"
+    "prepublishOnly": "make build",
+    "storybook": "start-storybook -p 3000",
+    "build-storybook": "build-storybook",
+    "chromatic": "npx chromatic --exit-zero-on-changes"
```
[the remark this answers is not in the document]
Comment (taylrj) [reply packages/react-components/package.json:15]: I get it now. 
According to the [ref](https://www.chromatic.com/docs/circleci#re-run-failed-builds-after-verifying-ui-test-results), builds that contain visual changes need to be [verified](https://www.chromatic.com/docs/test#verify-ui-changes). They will fail if you are not using `--exit-zero-on-changes`.

File: .circleci/config.yml [hunk 5]
```diff
@@ -130,6 +130,18 @@ commands:
...
+  deploy-storybook:
+    description: "Deploy storybook on chromatic"
+    steps:
+      - run:
+          name: 'run chromatic cli in react-components'
+          command: |
+            echo ${TEST}
+            echo ${CHROMATIC_TOKEN}
+            echo ${CHROMATIC_PROJECT_TOKEN}
+            printenv
```
Comment (taylrj) [inline .circleci/config.yml:142]: Question: why do we need to echo the environment variables and `printenv` before `yarn chromatic`?

File: packages/react-components/src/icon/topic.stories.js (hunk 2 above)
Comment (Aylie-Chou) [reply packages/react-components/src/icon/topic.stories.js]: updated. [a9edf8f](https://github.com/twreporter/twreporter-npm-packages/pull/262/commits/a9edf8ff81eaaa9c3f205688f698211eb62a262a)

File: .circleci/config.yml (hunk 3 above)
Comment (Aylie-Chou) [reply .circleci/config.yml:202]: updated. [a9edf8f](https://github.com/twreporter/twreporter-npm-packages/pull/262/commits/a9edf8ff81eaaa9c3f205688f698211eb62a262a)

File: .circleci/config.yml (hunk 5 above)
Comment (Aylie-Chou) [reply .circleci/config.yml]: updated. [a9edf8f](https://github.com/twreporter/twreporter-npm-packages/pull/262/commits/a9edf8ff81eaaa9c3f205688f698211eb62a262a)

File: packages/react-components/src/button/components/pillButton.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  const { color, bgColor, hoverColor, hoverBgColor } = themeFunc(
+    theme,
+    disabled
+  )
+  const { fontSize, padding, iconSize } = getSizeStyle(size)
+  return (
+    <ButtonContainer
+      type={type}
+      padding={padding}
+      color={color}
```
[the remark this answers is not in the document]
Comment (Aylie-Chou) [reply packages/react-components/src/button/components/pillButton.js:63]: After discussion with designer, the spec of button would have to be modified.
I would apply the change in futher commit 🙏

File: .circleci/config.yml (hunk 3 above)
Comment (Aylie-Chou) [reply .circleci/config.yml:202]: If we want to add ui review in PR review stage, the branch filter should accept all branch without master, staging and release.
updated [bdc9f19](https://github.com/twreporter/twreporter-npm-packages/pull/262/commits/bdc9f192d54759cae7db60c912e8751248f993f0).

File: packages/react-components/src/button/components/pillButton.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Aylie-Chou) [reply packages/react-components/src/button/components/pillButton.js:63]: updated [f814b0e](https://github.com/twreporter/twreporter-npm-packages/pull/262/commits/f814b0e6c61a27c098f45e28300e28783ddabaa3)

File: .circleci/config.yml [hunk 7]
```diff
@@ -135,7 +135,7 @@ commands:
     steps:
       - run:
           name: 'run chromatic cli in twreporter-npm-packages'
-          command: yarn chromatic --project-token=${CHROMATIC_TOKEN} --exit-zero-on-changes
```
Comment (taylrj) [inline .circleci/config.yml]: Question: why would we like to remove `--exit-zero-on-changes` ?

According to the documentation, it [states](https://www.chromatic.com/docs/circleci#command-exit-code-for-required-checks):

`If you are using pull request statuses as required checks before merging, you may not want your Circle CI job to fail if test snapshots render without errors (but with changes). To achieve this, pass the flag --exit-zero-on-changes to the chromatic command, and your job will continue in such cases.`

Does it meet our use case ?

Comment (Aylie-Chou) [reply .circleci/config.yml]: Yes, it meets our use case.
`--exit-zero-on-changes` flag would be add in package.json `chromatic` script.

**Comment by Aylie-Chou** [comment]
As previous discussion, remove ci config for this patch.
cc @taylrj

**Comment by taylrj** [comment]
Nice work ☄️.💥

As previous discussion, because it is closely related to the work that @duidae is doing, I'd like to invite @duidae to review this PR before it will be merged.

**Review by taylrj — APPROVED** [review]

**Pull request merged**